src/minio/core/policy_builder.py

```diff
@@ -370,4 +390,21 @@ def _add_object_permission(self, resource_arn: str, action: PolicyAction) -> Non
 
     def _create_list_bucket_prefixes(self, normalized_path: str) -> list[str]:
         """Create list of prefixes for ListBucket operations."""
-        return [normalized_path, f"{normalized_path}/*"]
+        prefixes = [normalized_path, f"{normalized_path}/*"]
+
+        # For governance paths with wildcards, also add parent directory access
+        # This allows users to navigate to paths like users-sql-warehouse/tgu2/u_tgu2__*
+        # by granting access to the parent directory users-sql-warehouse/tgu2
```

Comment:
Doesn't this allow the user to read / write to users-sql-warehouse/tgu2, defeating the purpose of the data governance prefixes?

Reply:
no - it allows users to view (list) the parent dir. It allows users to login and navigate to the namespaces dir.

Reply:
:+1:

Reply:
Why? It seems fine to me if they can't

Reply:
that's a different concept - from user's perspective, if we can provide a convenient method, we should do it. We already provide them the permission to delete/update files. Why not just allow them to easily navigate to path.

Reply:
yes - it applies to tenant sql files. So concept - we already provide group members permission to delete/update files. This just allows them to navigate to the dir.

Reply:
Ok, I disagree with this and think it's not necessary and not helpful, but I don't feel strongly enough about it to argue further